### PR TITLE
EV-20:  Refactor Contact Us page in profile

### PR DESCRIPTION
**Pull request by MelenkoOleksandr**
### Jira

* [EV-20](https://chita.atlassian.net/browse/EV-20)

### Description
Redesigned Contact Us Form(Report form) in profile. Added new components for form fields and rewriten on the new lib react-final-form 

### Checklist

- [x] Includes a link to JIRA ticket and all related PR (API dependencies etc.)
- [x] Includes a brief description of what has been changed
- [x] Test coverage for **new** code doesn't decrease the existing test coverage
- [x] Lint errors/warnings fixed
- [x] New code is documented if necessary
- [x] Dev Testing has been successfully finished
- [x] Reviewed by 2 developer on your team
- [x] Reviewed by 1 expert

### CODE REVIEW

File: src/components/ReportPage/ReportPage.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import ReportForm from "../ReportForm/ReportForm";
+import "./ReportPage.scss";
+
+const ReportPage = ({ handleReportFormSubmit }) => {
+  const validate = values => {
+    const errors = {};
+    const requiredFields = ["email", "description"];
+    if (maxLength30(values.title)) {
+      errors.title = "Title should be less 30 symbols";
+    }
```
Comment (ghost) [inline src/components/ReportPage/ReportPage.jsx]: take out to validators or helpers

File: src/components/shared/FormInput/FormInput.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  meta: { error, touched, invalid },
+  ...restProps
+}) => (
+  <div className="form-group__input">
+    <input
+      value={value}
+      onChange={onChange}
+      onBlur={onBlur}
+      className={`form-input ${className} ${
+        invalid && touched ? " form-input__error" : ""
```
Comment (ghost) [inline src/components/shared/FormInput/FormInput.jsx]: KISS

File: src/components/shared/FormTextArea/FormTextArea.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  ...restProps
+}) => (
+  <div className="form-group__textarea">
+    <textarea
+      value={value}
+      onChange={onChange}
+      onBlur={onBlur}
+      className={`form-textarea ${className} ${
+        error && touched ? " form-textarea__error" : ""
+      }`}
```
Comment (ghost) [inline src/components/shared/FormTextArea/FormTextArea.jsx]: KISS

**Review by ghost — CHANGES_REQUESTED** [review]
needs editing

File: src/components/ReportPage/ReportPage.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  };
+  return (
+    <div className="report__container">
+      <h3 className="report-container__title">{REPORT_FORM_TITLE}</h3>
+      <Form
+        onSubmit={onHandleReportFormSubmit}
+        validate={validate(
+          ["email", "description"],
+          [{ field: "title", maxLen: 30 }],
+        )}
```
Comment (ghost) [inline src/components/ReportPage/ReportPage.jsx]: i suggest moveing it from return(...)

File: src/components/helpers/validators/email-address-validator.js [hunk 5]
```diff
@@ -1,9 +1,8 @@
 const isValidEmail = email => {
-  const errors = {};
-  if (email && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email)) {
-    errors.email = "Invalid email address";
+  if (!email || !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email)) {
```
Comment (ghost) [inline src/components/helpers/validators/email-address-validator.js]: use constatns for regExp

File: src/containers/RegisterContainer/RegisterBindAccountContainer.js [hunk 6]
```diff
@@ -1,19 +1,7 @@
...
-  const requiredFields = ["password", "email", "type"];
-
-  return {
-    ...fieldIsRequired(values, requiredFields),
-    ...isValidEmail(values.email),
-  };
-};
+import { validate } from "../../components/helpers/validateHelper";
 
 const mapStateToProps = state => {
```
Comment (ghost) [inline src/containers/RegisterContainer/RegisterBindAccountContainer.js:6]: try to destructure state

**Review by ghost — DISMISSED** [review]

File: src/containers/RegisterContainer/RegisterBindAccountContainer.js (hunk 6 above)
Comment (MelenkoOleksandr) [reply src/containers/RegisterContainer/RegisterBindAccountContainer.js:6]: This is not my component .My task was to create universal vallidation function and change in all places.

File: src/components/helpers/validators/email-address-validator.js (hunk 5 above)
Comment (MelenkoOleksandr) [reply src/components/helpers/validators/email-address-validator.js]: Moved to helpersConstants

**Pull request closed without merging**